src/Symfony/Component/Lock/Tests/FactoryTest.php

```diff
@@ -23,6 +26,22 @@
 class FactoryTest extends TestCase
 {
     public function testCreateLock()
+    {
+        $store = $this->getMockBuilder([PersistStoreInterface::class, BlockingStoreInterface::class, ExpiringStoreInterface::class])->getMock();
+        $logger = $this->getMockBuilder(LoggerInterface::class)->getMock();
+        $factory = new Factory($store);
+        $factory->setLogger($logger);
+
+        $lock = $factory->createLock('foo');
+
+        $this->assertInstanceOf(LockInterface::class, $lock);
+    }
+
+    /**
+     * @group legacy
+     * @expectedDeprecation "Symfony\Component\Lock\StoreInterface" is deprecated since Symfony 4.4 and has been splitted into "Symfony\Component\Lock\PersistStoreInterface", "Symfony\Component\Lock\ExpiringStoreInterface", "Symfony\Component\Lock\BlockingStoreInterface".
+     */
+    public function testCreateLockLegacy()
```

Comment:
this legacy test does not make sense. The Factory does not contain any BC layer.